lib/MusicBrainz/Server/Edit/Medium/Edit.pm

```diff
@@ -316,14 +328,21 @@ sub build_display_data
             $data->{changed_mbids} = 1;
         }
 
-        $data->{artist_credit_changes} = [
+        my @artist_credit_changes =
+            map { {
+                change_type => $_->[0],
+                old_track => $_->[1] eq '' ? undef : $_->[1],
+                new_track => $_->[2],
+            } }
             grep {
                 ($_->[1] && hash_artist_credit_without_join_phrases($_->[1]->artist_credit))
                     ne
                 ($_->[2] && hash_artist_credit_without_join_phrases($_->[2]->artist_credit))
             }
             grep { $_->[0] ne '-' }
-            @$tracklist_changes ];
+            @$tracklist_changes;
+
+        $data->{artist_credit_changes} = \@artist_credit_changes;
```

Comment:
Not sure this change is necessary, since `$data->{artist_credit_changes} = [ ... ]` was already easy to read and the variables aren't used elsewhere? Same with recording_changes below. (Sorry to nitpick again, I just don't like noisy diffs. :P)